File: 100-same-tree/same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:

        def dfsSol(p, q):
            if not p and not q:
                return True
            if p and q and p.val == q.val:
                return dfsSol(p.left, q.left) and dfsSol(p.right, q.right)
            else:
                return False

        return dfsSol(p,q)

File: 100-same-tree/same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
    function bfsSol(p, q) {
    const p1 = [];  
    const p2 = []; 

    p1.push(p);
    p2.push(q);

    while (p1.length > 0 && p2.length > 0) {
        const nodeP = p1.shift();
        const nodeQ = p2.shift();

        if (nodeP === null && nodeQ === null) {
            continue;
        }
        if (nodeP === null || nodeQ === null || nodeP.val !== nodeQ.val) {
            return false;
        }

        p1.push(nodeP.left);
        p1.push(nodeP.right);
        p2.push(nodeQ.left);
        p2.push(nodeQ.right);
    }

    return true;
}
    return bfsSol(p,q)
};